### DISCUSSION TITLE
Build external URLs when behind a reverse proxy

### DISCUSSION THREAD

**Discussion by muckelba** [pocketbase/pocketbase#3115] [Q&A] [answered]
Hi, i just deployed my sveltekit+pocketbase application on my server using docker-compose. Both containers are exposed via traefik on two different subdomains. In my svelte code i access the pocketbase container by referencing its internal dns name (`pocketbase`). But when i now try to show pictures in my webapp, i get 404's because `pb.files.getUrl()` is also using the internal name and no client on the normal internet is able to access `pocketbase`. 

Is there any config option to either tell pocketbase itself or the js-sdk to build URLs with the actual domain used by my reverse proxy? Or should i build my own function that concatenates external domain, document id and filename together?

Thanks in advance :)

**Comment by ganigeorgiev** [comment, marked as answer]
`getUrl()` uses the base url with which the SDK was initialized.
Currently you'll have to create your own helper if you want to use a different url.
For how to construct the final url you can check https://pocketbase.io/docs/files-handling/#file-url (the `collectionId` and the record `id` fields can be retrieved from the record model itself; the final part is the name of the file).

**Reply by muckelba** [reply]
Thanks for the fast reply! It's okay, i'll build my own helper :)
